src/lib.rs

```diff
@@ -896,14 +901,17 @@ pub struct Object {
     pub visible: bool,
     pub shape: ObjectShape,
     pub properties: Properties,
+    pub template: Option<Box<Template>>,
```

Comment:
What's the reason for a box here? I don't think the size of a template is great enough for it to be worth doing from a memory perspective

Reply:
Isn't it necessary because of the recursive use of `Object` in `Template`? Can a `struct Object` include an `Option<Object>` as one of its members?